flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -859,7 +860,7 @@ private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint) thro
 			catch (Exception e1) {
 				// abort the current pending checkpoint if we fails to finalize the pending checkpoint.
 				if (!pendingCheckpoint.isDiscarded()) {
-					failPendingCheckpoint(pendingCheckpoint, CheckpointFailureReason.FINALIZE_CHECKPOINT_FAILURE, e1);
+					failPendingCheckpoint(pendingCheckpoint, CheckpointFailureReason.FINALIZE_CHECKPOINT_FAILURE, e1, null);
```

Comment:
ditto: add an overloaded `failPendingCheckpoint` method?

Reply:
I already have introduced a new `handleCheckpointException` method with 3 parameters, and it requires to do a type casting on `null` when introducing another one and invoking with `null` as the last parameter, which I _thought_ to be not recommended. But after a [search and check](https://coderanch.com/t/382683/java/Casting-null-object) I believe it's fine for casting null to help compiler choose between overloading methods, so yes let me do this.